### DISCUSSION TITLE
How to retry a response that on 502?

### DISCUSSION THREAD

**Discussion by hackerliang** [mitmproxy/mitmproxy#5136] [Q&A] [answered]
Hello everyone,
First of all thanks for this awesome project.

I want to retry a request that the response on 502.

That means, e.g.

The browser makes a request through mitmproxy, but the server response returns a 502 Bad Gateway, then mitmproxy do not return this 502 response to the browser, instead, mitmproxy keep resending the original request to the server unless the server response a none 502 response, then mitmproxy return the none 502 response to the browser.

During this whole process, the browser does not know that the server has ever returned a 502.

**Comment by mhils** [comment, marked as answer]
I think your question is a duplicate of #5023.
